game/i18n.hh

```diff
@@ -2,26 +2,130 @@
 
 #include <boost/locale.hpp>
 #include <iostream>
+#include <string>
+#include <map>
+
+#include "configuration.hh"
 #include "fs.hh"
 
 #define _(x) boost::locale::translate(x).str()
 #define translate_noop(x) x
 
 class TranslationEngine {
 public:
-	TranslationEngine(const char *package) {
-		boost::locale::generator gen;
-		gen.add_messages_path(getLocaleDir().string());
-		gen.add_messages_domain(package);
-		try {
-			std::locale::global(gen(boost::locale::util::get_system_locale(true)));
-		} catch (...) {
-			std::clog << "locale/warning: Unable to detect locale, will try to fallback to en_US.UTF-8" << std::endl;
-			std::locale::global(gen("en_US.UTF-8"));
-		}
+	TranslationEngine(const char *package) : m_package(package) {
+
+		initializeAllLanguages();
+		populateLanguages(GetAllLanguages());
+
+		std::clog << "locale/debug: Checking if language is set in config." << std::endl;
+
+		auto prefValue = config["game/language"].getEnumName();
+		auto lang = getLanguageByHumanReadableName(prefValue);
+
+		setLanguage(lang);
 	};
 	static bool enabled() {
 		return true;
 	};
 
+	static ConfigItem& translationConfig() {
```

Comment:
you are creating a singleton; is that really *needed* why not just returning the configIten by value?

Reply:
This method gets called from the config WriteConfig method. We don't have any reference to TranslationEngine there. Currently looking into it if it's necessairy

Reply:
Removed it as it wasn't necessairy